Purchases/Public/RCEntitlementInfo.m

```diff
@@ -0,0 +1,177 @@
+    } else if ([store isEqualToString:@"promotional"]) {
+        return RCPromotional;
+    }
+    return RCUnknownStore;
+}
+
+- (NSString *)description
+{
+    NSMutableString *description = [NSMutableString stringWithFormat:@"<%@: ", NSStringFromClass([self class])];
+    [description appendFormat:@"identifier=%@,\n", self.identifier];
+    [description appendFormat:@"isActive=%d,\n", self.isActive];
+    [description appendFormat:@"willRenew=%d,\n", self.willRenew];
+    [description appendFormat:@"periodType=%li,\n", (long) self.periodType];
+    [description appendFormat:@"latestPurchaseDate=%@,\n", self.latestPurchaseDate];
+    [description appendFormat:@"originalPurchaseDate=%@,\n", self.originalPurchaseDate];
+    [description appendFormat:@"expirationDate=%@,\n", self.expirationDate];
+    [description appendFormat:@"store=%li,\n", (long) self.store];
+    [description appendFormat:@"productIdentifier=%@,\n", self.productIdentifier];
+    [description appendFormat:@"isSandbox=%d,\n", self.isSandbox];
+    [description appendFormat:@"unsubscribeDetectedAt=%@,\n", self.unsubscribeDetectedAt];
+    [description appendFormat:@"billingIssueDetectedAt=%@,\n", self.billingIssueDetectedAt];
+    [description appendString:@">"];
+    return description;
+}
+
+- (BOOL)isEqual:(id)other
+{
+    if (other == self)
+        return YES;
+    if (!other || ![[other class] isEqual:[self class]])
+        return NO;
+
+    return [self isEqualToInfo:other];
+}
+
+- (BOOL)isEqualToInfo:(RCEntitlementInfo *)info
+{
+    if (self == info)
+        return YES;
+    if (info == nil)
+        return NO;
+    if (self.identifier != info.identifier && ![self.identifier isEqualToString:info.identifier])
+        return NO;
+    if (self.isActive != info.isActive)
+        return NO;
+    if (self.willRenew != info.willRenew)
+        return NO;
+    if (self.periodType != info.periodType)
+        return NO;
+    if (self.latestPurchaseDate != info.latestPurchaseDate && ![self.latestPurchaseDate isEqualToDate:info.latestPurchaseDate])
+        return NO;
+    if (self.originalPurchaseDate != info.originalPurchaseDate && ![self.originalPurchaseDate isEqualToDate:info.originalPurchaseDate])
+        return NO;
+    if (self.expirationDate != info.expirationDate && ![self.expirationDate isEqualToDate:info.expirationDate])
+        return NO;
+    if (self.store != info.store)
+        return NO;
+    if (self.productIdentifier != info.productIdentifier && ![self.productIdentifier isEqualToString:info.productIdentifier])
+        return NO;
+    if (self.isSandbox != info.isSandbox)
+        return NO;
+    if (self.unsubscribeDetectedAt != info.unsubscribeDetectedAt && ![self.unsubscribeDetectedAt isEqualToDate:info.unsubscribeDetectedAt])
+        return NO;
+    if (self.billingIssueDetectedAt != info.billingIssueDetectedAt && ![self.billingIssueDetectedAt isEqualToDate:info.billingIssueDetectedAt])
+        return NO;
+    return YES;
+}
+
+- (NSUInteger)hash
```

Comment:
Are you using `EntitlementInfos` as a dictionary key? If not you probably don't need to implement this do you?

Reply:
Nope. I autogenerated this and isEqual. Removing!